src/ucp/core/ucp_worker.c

```diff
@@ -218,10 +218,73 @@ ucp_to_uct_wakeup_events(ucp_wakeup_event_t ucp_events)
     return uct_events;
 }
 
+
+static void
+uct_to_ucp_error_proxy(void *arg, uct_ep_h uct_ep, ucs_status_t status)
+{
+    ucp_worker_h       worker           = (ucp_worker_h)arg;
+    ucp_ep_h           ucp_ep           = NULL;
+    uint64_t           dest_uuid        = 0ul;
+    ucp_ep_h           ucp_ep_iter;
+    khiter_t           ucp_ep_errh_iter;
+    ucp_err_handler_t  err_handler;
+    ucp_rsc_index_t    lane, n_lanes, failed_lane;
+
+    /* TODO: need to optimize uct_ep -> ucp_ep lookup */
+    kh_foreach(&worker->ep_hash, dest_uuid, ucp_ep_iter, {
+        for (lane = 0; (lane < ucp_ep_num_lanes(ucp_ep_iter)); ++lane) {
+            if ((uct_ep == ucp_ep_iter->uct_eps[lane]) && dest_uuid) {
+                ucp_ep = ucp_ep_iter;
+                failed_lane = lane;
+                goto found_ucp_ep;
+            }
+        }
+    });
+
+    ucs_fatal("No uct_ep_h %p associated with ucp_ep_h on ucp_worker_h %p",
+              uct_ep, worker);
+    return;
+
+found_ucp_ep:
+    ucp_ep_config(ucp_ep)->status = status;
+
+/* Destroy all lanes excepting failed one since ucp_ep becomes unusable as well */
+    for (lane = 0, n_lanes = ucp_ep_num_lanes(ucp_ep); lane < n_lanes; ++lane) {
+        if ((lane != failed_lane) && ucp_ep->uct_eps[lane]) {
+            uct_ep_destroy(ucp_ep->uct_eps[lane]);
+            ucp_ep->uct_eps[lane] = NULL;
+            ucp_ep_config(ucp_ep)->key.num_lanes--;
+        }
+    }
+    ucs_assert(ucp_ep_config(ucp_ep)->key.num_lanes == 1);
+
+    /* Move failed lane at 0 index */
+    if (failed_lane != 0) {
+        ucp_ep->uct_eps[0] = ucp_ep->uct_eps[failed_lane];
+        ucp_ep->uct_eps[failed_lane] = NULL;
+    }
+
+    /* Redirect all lanes to failed one */
+    ucp_ep->am_lane                        =
+    ucp_ep_config(ucp_ep)->key.am_lane     =
```

Comment:
this cannot be done - breaks all other endpoints using same config.
need to create a new configuration key and allocate index using ucp_worker_get_ep_config()